torchreid/data/datasets/image/classification.py

```diff
@@ -107,6 +107,13 @@ def __init__(self, data_provider, mode='train', dataset_id=0, load_masks=False,
 
         super().__init__(train, query, gallery, mode=mode, **kwargs)
 
+
+        # restore missing classes in train
```

Comment:
Is it for the case of early training stages when we may not have all classes in training?

Reply:
Yes. This allows run training without having some classes at all, while at he moment such training fails.